integration/test_utils.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"errors"
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/kubelet/apis/cri"
+	"k8s.io/kubernetes/pkg/kubelet/apis/cri/v1alpha1/runtime"
+	"k8s.io/kubernetes/pkg/kubelet/remote"
+
+	"github.com/kubernetes-incubator/cri-containerd/pkg/util"
+)
+
+const (
+	sock    = "/var/run/cri-containerd.sock"
+	timeout = 1 * time.Minute
+)
+
+var (
+	runtimeService cri.RuntimeService
+	imageService   cri.ImageManagerService
+)
+
+func init() {
+	var err error
+	runtimeService, err = remote.NewRemoteRuntimeService(sock, timeout)
+	if err != nil {
+		glog.Exitf("Failed to create runtime service: %v", err)
+	}
+	imageService, err = remote.NewRemoteImageService(sock, timeout)
+	if err != nil {
+		glog.Exitf("Failed to create image service: %v", err)
+	}
+}
+
+// Opts sets specific information in pod sandbox config.
+type PodSandboxOpts func(*runtime.PodSandboxConfig)
+
+// PodSandboxConfig generates a pod sandbox config for test.
+func PodSandboxConfig(name, ns string, opts ...PodSandboxOpts) *runtime.PodSandboxConfig {
+	config := &runtime.PodSandboxConfig{
+		Metadata: &runtime.PodSandboxMetadata{
+			Name: name,
+			// Using random id as uuid is good enough for local
+			// integration test.
+			Uid:       util.GenerateID(),
+			Namespace: ns,
+		},
+		Linux: &runtime.LinuxPodSandboxConfig{},
+	}
+	for _, opt := range opts {
+		opt(config)
+	}
+	return config
+}
+
+// Eventually waits for f to return nil, it checks every period, and
+// returns error if timeout exceeds.
+func Eventually(f func() (bool, error), period, timeout time.Duration) error {
```

Comment:
It will be better if `func() (bool, error)` is defined as a type, and given an explanation.

Reply:
Will do.